include/adm/elements/audio_channel_format.hpp

```diff
@@ -360,29 +360,46 @@ namespace adm {
   }
 
   template <typename BlockFormat>
-  void AudioChannelFormat::assignId(BlockFormat &blockFormat) {
-    auto id = blockFormat.template get<AudioBlockFormatId>();
-
-    TypeDescriptor typeDescriptor;
-    AudioBlockFormatIdValue value;
-    AudioBlockFormatIdCounter counter;
-
-    if (isUndefined(id)) {
-      typeDescriptor = get<TypeDescriptor>();
-      value = AudioBlockFormatIdValue(
-          get<AudioChannelFormatId>().get<AudioChannelFormatIdValue>().get());
-      counter = AudioBlockFormatIdCounter(1u);
-    } else {
-      typeDescriptor = id.template get<TypeDescriptor>();
-      value = id.template get<AudioBlockFormatIdValue>();
-      counter = id.template get<AudioBlockFormatIdCounter>();
-    }
+  void AudioChannelFormat::assignId(BlockFormat &blockFormat,
+                                    BlockFormat *previousBlock) {
+    auto thisId = blockFormat.template get<AudioBlockFormatId>();
+
+    auto expectedTypeDescriptor = get<TypeDescriptor>();
+    auto expectedValue = AudioBlockFormatIdValue(
+        get<AudioChannelFormatId>().get<AudioChannelFormatIdValue>().get());
+
+    if (isUndefined(thisId)) {
+      AudioBlockFormatIdCounter counter;
+      if (previousBlock) {
+        auto prevId = previousBlock->template get<AudioBlockFormatId>();
+        counter = AudioBlockFormatIdCounter(
+            prevId.template get<AudioBlockFormatIdCounter>().get() + 1);
+      } else {
+        counter = AudioBlockFormatIdCounter(1u);
+      }
+
+      blockFormat.set(
+          AudioBlockFormatId(expectedTypeDescriptor, expectedValue, counter));
 
-    while (idUsed<BlockFormat>(
-        AudioBlockFormatId(typeDescriptor, value, counter))) {
-      ++counter;
+    } else {
+      auto thisTypeDescriptor = thisId.template get<TypeDescriptor>();
+      if (thisTypeDescriptor != expectedTypeDescriptor)
+        throw std::runtime_error("Invalid ID - incorrect type descriptor");
+
+      auto thisValue = thisId.template get<AudioBlockFormatIdValue>();
+      if (thisTypeDescriptor != expectedTypeDescriptor)
```

Comment:
This might be a copy & paste error? 
It probably should be `if (thisValue != expectedValue)` ?

Reply:
Yes, definitely a mistake :man_facepalming: Good spot!